### PR TITLE
Docs about how to generate config for act runner with docker and setup it with docker-compose

### PR DESCRIPTION
**Pull request by thezzisu**
In this pull request, the following changes are addressed:

- State user should create `config.yaml` before start container to avoid errors.
- Provided instructions to deploy runners using docker compose.

### CODE REVIEW

**Comment by thezzisu** [comment]
The motivation of this PR is that I have wasted half an hour debugging my network after seen the `Error: EOF` log when running the runner container. However, the cause is I created a empty `config.yaml`. I think the documentation should state this to avoid misunderstandings.

File: docs/content/doc/usage/actions/act-runner.en-us.md [hunk 1]
```diff
@@ -82,6 +82,12 @@ When you are using the docker image, you can specify the configuration file by u
 docker run -v $(pwd)/config.yaml:/config.yaml -e CONFIG_FILE=/config.yaml ...
 ```
 
+You need to create config.yaml before running the container, to avoid `Error: EOF` errors. Neither use the method mentioned above, or just run the following command:
```
Comment (lunny) [inline docs/content/doc/usage/actions/act-runner.en-us.md]: config.yaml is not a necessary if you are using a default setting. But if you put an empty file, then what should happen?

Comment (thezzisu) [reply docs/content/doc/usage/actions/act-runner.en-us.md]: image
As I mentioned, the result is `Error: EOF` which is very confusing.

Comment (thezzisu) [reply docs/content/doc/usage/actions/act-runner.en-us.md]: Meanwhile, docker mount/volume will create a directory when the mount source is not existing. So maybe creating a `config.yaml` in advance is the best practice.

Comment (lunny) [reply docs/content/doc/usage/actions/act-runner.en-us.md]: Yes, maybe you should create an empty config file but not a folder?

Comment (thezzisu) [reply docs/content/doc/usage/actions/act-runner.en-us.md]: Yes, but as the screenshot above create a empty config file will cause the program to crash with a confusing error message. After roughly viewed the source code, I think the runner requires a well formated yaml config to work, but a empty yaml file just cannot make the parser happy.

Comment (thezzisu) [reply docs/content/doc/usage/actions/act-runner.en-us.md]: Thus, it's neede to using generate-config to generate a working config first.
EDT: In the screenshot above, `$(pwd)/config.yaml` is an already created empty file.

**Comment by wolfogre** [comment]
My opinions:

- Empty `config.yaml` should be acceptable, the EOF error surprises me. I will I'm going to figure it out and fix it.
- I think "Make sure that the file is mounted into the container as a volume" is clear enough. That meas you get the file already, of cause "You need to create config.yaml before running the container".
  - image
- `docker run --entrypoint="" --rm -it gitea/act_runner:latest act_runner generate-config > config.yaml` is a good idea, it could be ”How to generate config file with a docker container“
  - image
- "Set up the runner using docker compose" looks great to me.

So my suggestions are:
- Remove lines with "Error EOF"
- Keep `docker run --entrypoint="" --rm -it gitea/act_runner:latest act_runner generate-config > config.yaml` and "Set up the runner using docker compose"

**Comment by thezzisu** [comment]
@wolfogre thanks for the suggestion. Also, I've opened a related issue on act_runner at https://gitea.com/gitea/act_runner/issues/240.

**Comment by wolfogre** [comment]
Update: https://gitea.com/gitea/act_runner/pulls/241

File: docs/content/doc/usage/actions/act-runner.en-us.md [hunk 2]
```diff
@@ -76,6 +76,12 @@ The default configuration is safe to use without any modification, so you can ju
 ./act_runner --config config.yaml [command]
 ```
 
+You could also generate generate config file with docker:
```
Comment (wolfogre) [inline docs/content/doc/usage/actions/act-runner.en-us.md]:
```suggestion
You could also generate config file with docker:
```

**Review by wolfogre — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**